packages/react-components/src/components/gif.tsx

```diff
@@ -18,6 +18,8 @@ export const getColor = () => GRID_COLORS[Math.round(Math.random() * (GRID_COLOR
 
 const hoverTimeoutDelay = 200
 
+const WithLink = (props: any) => (props.href ? <a href={props.href} {...props} /> : <div {...props} />)
```

Comment:
@thighhighcrocs `With` reminds me of HoCs, maybe we can just call it `Container`? I added types:
```typescript
type ContainerProps = HTMLProps<HTMLElement> & { href: string }
const Container = (props: ContainerProps) =>
    props.href ? <a {...(props as HTMLProps<HTMLAnchorElement>)} /> : <div {...(props as HTMLProps<HTMLDivElement>)} />
```

Reply:
I like it!

Reply:
@thighhighcrocs sweet! not sure how the types would work in `Preact`, if they have a `HTMLProps<>` or not. Also for the `href` I just passed `gif.url` or `''`